src/preprocess.py

```diff
@@ -262,14 +285,15 @@ class InvertedIndex:
     """
     itype: int
     index: Union[
-        OrdDict[str, List[int]], # itype == 1
+        OrdDict[str, OrdDict[int, bool]], # itype == 1
         OrdDict[str, OrdDict[int, int]], # itype == 2
         OrdDict[str, OrdDict[int, List[int]]] # itype == 3
         ]
     mapping: OrdDict[int, str]
     stats: StatCollection
 
 # TODO: @Antoine-marchais : This part could be refactored with auxiliary function for lisibility.
+@timer
 def build_inverted_index(
```

Comment:
Still no factorization there, you deserved it : 
![image](https://user-images.githubusercontent.com/26746350/79766960-f9236b80-8328-11ea-99cd-7c1e5d426e3b.png)